app/gui/view/component-browser/component-list-panel/grid/src/entry/style.rs

```diff
@@ -5,63 +5,137 @@ use crate::prelude::*;
+        let path = path.into();
+        let value = style.get(path.clone());
+        frp::extend! { network
+            init <- source_();
+            color <- value.all_with(&init, move |data, _| {
+                data.color().map(|color| {
+                    let color = color::Lcha::from(color);
+                    Color::Arbitrary(color)
+                }).unwrap_or_else(|| {
+                    let intensity = match data.number() {
+                        Some(number) => number,
+                        None => {
+                            error!("Neither color nor intensity defined ({path}).");
+                            0.0
+                        }
+                    };
+                    Color::MainColorIntensity(intensity)
+                })
+            });
+            sampler <- color.sampler();
+        }
+        init.emit(());
+        sampler
+    }
+}
+
+
+
 // =============
 // === Style ===
 // =============
 
-// === Color Intensities ===
+// === Style Colors ===
 
-/// The intensities of various parts of Component Entry view. The actual color is computed by mixing
-/// the main groups color with the application background - see [`Colors`] for more information.
+/// The colors of various parts of the Component Entry view. The actual color can be computed by
+/// mixing the "main" group's color with the application background - see [`Color`] for more
+/// information.
 #[allow(missing_docs)]
 #[derive(Clone, Copy, Debug, Default, PartialEq, FromTheme)]
-pub struct ColorIntensities {
-    #[theme_path = "entry_theme::text::color_intensity"]
-    pub text:            f32,
-    #[theme_path = "entry_theme::background::color_intensity"]
-    pub background:      f32,
-    #[theme_path = "entry_theme::highlight::hover::color_intensity"]
-    pub hover_highlight: f32,
-    #[theme_path = "entry_theme::dimmed::color_intensity"]
-    pub dimmed:          f32,
+pub struct StyleColors {
+    #[theme_path = "entry_theme::text::color"]
+    #[accessor = "Color::accessor"]
+    pub text:            Color,
+    #[theme_path = "entry_theme::background::color"]
+    #[accessor = "Color::accessor"]
+    pub background:      Color,
+    #[theme_path = "entry_theme::highlight::hover::color"]
+    #[accessor = "Color::accessor"]
+    pub hover_highlight: Color,
     /// The more contrasting parts of the [icon](crate::icon::Any).
-    #[theme_path = "entry_theme::icon::strong_color_intensity"]
-    pub icon_strong:     f32,
+    #[theme_path = "entry_theme::icon::strong_color"]
+    #[accessor = "Color::accessor"]
+    pub icon_strong:     Color,
     /// The less contrasting parts of the [icon](crate::icon::Any).
-    #[theme_path = "entry_theme::icon::weak_color_intensity"]
-    pub icon_weak:       f32,
+    #[theme_path = "entry_theme::icon::weak_color"]
+    #[accessor = "Color::accessor"]
+    pub icon_weak:       Color,
+    #[theme_path = "entry_theme::dimmed"]
+    pub dimmed:          color::Rgba,
```

Comment:
Dimmed color can't use `Color` infrastructure sadly. So you can only set it to arbitrary color. You can't calculate it as a mix between the group's "main" color, because the dimmed color **is** the "main" color of the dimmed component group now. And all other colors of the dimmed component group are calculated, as usual, using `dimmed` as the "main" color.

Reply:
I don't understand this explanation, especially this part `You can't calculate it as a mix between the group's "main" color, because the dimmed color is the "main" color of the dimmed component group now.` - would you be so nice and explain it a little bit more verbose for me, pleasE?

Reply:
What the dimmed color is used for? It is used as the icon main color when the icon is disabled, right? If so, this needs to be mixed with background. In most cases we will use black here with transparency and wee need to inherit some of the "blueness" from the background.

Reply:
Not exactly, it is not related to icons, but rather to the whole component group. Icon colors are always mixed with the background.

So we have like 6 distinct colors for component groups. I call them "main" colors, all other colors in the component list panel are derived from them by mixing with the background. Previously dimmed component groups were duller, so the `dimmed` state parameter controlled the intensity of the "main" color. 

In a new design, all dimmed component groups share the same color that does not depend on the "main" color. So `dimmed` can be considered the 7th color of the component group, it is used as the "main" color for dimmed component groups, and their colors are derived from `dimmed` instead of the "main" color. I will extend the docs to describe it.

Reply:
Why cannot it work with `Color`? If we have `Color::MainColorIntensity` then it could work as before, if we have `Color::Arbitrary` it could work as now.

Reply:
Ah, yes, it can work with `Color`. I missed the critical detail that you should mix `main_color`, not `color_anim.value`. So it is implemented now.